braid/_braid.c

```diff
@@ -2727,9 +2833,27 @@ _braid_FRefine(braid_Core   core,
    {
       _braid_CoreElt(core, r_space) = 0;
    }
-
+      
+   /* If periodic, add refinement to ensure the required number of points */
+   if (periodic)
+   {
+      f_gupper++;
+      _braid_PeriodicAdjustRFactors(core, &r_npoints, &f_gupper);
+      f_gupper--;
+   }
+      
    /* Compute r_ilower and r_iupper */
-   MPI_Scan(&r_npoints, &r_iupper, 1, braid_MPI_INT, MPI_SUM, comm);
+   {
+      braid_Int  inbuf = r_npoints;
+
+      /* If periodic, adjust the r_iupper index values in the scan by adjusting
+       * the 'inbuf' value on the first time interval */
+      if ( periodic && (ilower == 0))
+      {
```

Comment:
I don't see why you subtract by rfactors[0] here, but I'm probably just missing something

Reply:
This is a trick of sorts so that the scan will return the correct r_iupper (refined iupper) values.  You need to write down an example to see why it works.  A simple one with two procs should do.  Suppose I am refining the periodic interval by 4, specified at rfactors[0], and suppose the number of refined points (r_npoints) on both processors is 10.  Then, processor 0 should have a refined interval of [-3, 6] and processor 1 should have [7, 16].  I can make this happen by simply decrementing the inbuf value on processor zero by 3 and then do the scan.

Reply:
Ahh...I see.  Thanks!